platform/mbed_retarget.cpp

```diff
@@ -826,8 +826,13 @@ void mbed_set_unbuffered_stream(std::FILE *_file) {
  */
 std::FILE *mbed_fdopen(FileHandle *fh, const char *mode)
 {
-    char buf[12]; /* :0x12345678 + null byte */
-    std::sprintf(buf, ":%p", fh);
+    // This is to avoid scanf(buf, ":%.4s", fh) and the bloat it brings.
+    char buf[2 + sizeof(fh) + 1]; /* :(pointer) + null byte */
```

Comment:
I'm not aware of the internals of stdlib between the call to `std::fopen` and its call to `_open`, I'll take @pan-'s word that none of it does something that will read off the end of that buffer. The `strcmp` calls inside of `_open` are safe since none of them will read past the : at the start of `buf`.